source/common/common/base64.cc

```diff
@@ -24,118 +27,149 @@ static const unsigned char REVERSE_LOOKUP_TABLE[256] = {
     64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
     64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64};
 
-std::string Base64::decode(const std::string& input) {
-  if (input.length() % 4 || input.empty()) {
-    return EMPTY_STRING;
+constexpr char URL_CHAR_TABLE[] =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";
+
+constexpr unsigned char URL_REVERSE_LOOKUP_TABLE[256] = {
```

Comment:
How was this table generated? Did you copy from the same place as above? If so an you comment here also?